### ISSUE TITLE
Feature request: make `run --forget` a `forget-results` command

### ISSUE THREAD

**Issue by me-and** [mhagger/git-test#11]
It took me a while to work out how to get git-test to forget the results for a single commit, despite it being in the README, because I was expecting it to be part of `forget-results`, not `run`. I think it being part of `run` is confusing, and I'd like to request it be added to `forget-results`' functionality.

I'd expect the syntax to be something like `git forget-results [<commit>...]`, where if `<commit>...` is not specified the command works as at present, while if it is specified, it works the same as `git run --forget <commit>...`.

There's a small-but-greater-than-zero chance I'll manage a PR myself for this, but given the other projects currently on my plate I don't want to promise anything.

**Comment by mhagger** [comment]
I agree that it's harder than necessary to find the existing `--forget` option. OTOH I'm worried that tacking it on to `forget-results` would be dangerous if the `forget-results` default is to discard *all* results. It seems like it would be too easy to discard all of your results by accident.

If we go this route, then I think I'd propose that we make `git test forget-results` without an argument do *nothing* (but emit an error message), and require that an option like `--all` be passed if the user really wants to forget all results for the test. This of course would break backwards compatibility, but it's an obscure command in the first place, and it doesn't seem like all that many people are using this tool, so we could probably get away with it.
